plugins/discovery-gce/src/main/java/org/elasticsearch/cloud/gce/util/Access.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.cloud.gce.util;
+
+import org.elasticsearch.SpecialPermission;
+
+import java.io.IOException;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+
+/**
+ * GCE's http client changes access levels. Additionally remote calls need SocketPermissions for 'connect'.
+ * This class wraps these operations in {@link AccessController#doPrivileged(PrivilegedAction)} blocks.
+ */
+public class Access {
```

Comment:
I specifically name this one differently, because it is used for other permissions than just socket permissions. The GCE plugin requires additional permissions.

```
// needed because of problems in gce
  permission java.lang.RuntimePermission "accessDeclaredMembers";
  permission java.lang.RuntimePermission "setFactory";
  permission java.lang.reflect.ReflectPermission "suppressAccessChecks";
```

And some of the [usages](https://github.com/tbrooks8/elasticsearch/blob/migrate_connect/plugins/discovery-gce/src/main/java/org/elasticsearch/cloud/gce/GceMetadataService.java#L75) of this class are unrelated to socket permissions. These usages preexist my PR.

In the other plugins I named the helper classes SocketAccess because that is the only thing I use them for.